src/mdacli/logger.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+#
+# Copyright (c) 2021 Authors and contributors
+#
+# Released under the GNU Public Licence, v2 or any higher version
+# SPDX-License-Identifier: GPL-2.0-or-later
+"""Logging."""
+
+import contextlib
+import logging
+from logging.handlers import RotatingFileHandler
+
+from mdacli.colors import Emphasise
+
+
+DEBUGFORMATTER = '[{levelname}] {filename}:{name}:{funcName}:{lineno}:'
+DEBUGFORMATTER += '{message}'
+"""Debug file formatter."""
+
+INFOFORMATTER = '{message}'
+"""Log file and stream output formatter."""
+
+logger = logging.getLogger(__name__)
+
+
+@contextlib.contextmanager
+def setup_logging(logfile=None, debug=False):
+    """
+    Create a logging environment.
+
+    Parameters
+    ----------
+    logfile : str
+        Name of the log file
+    debug : bool
+        Display debug logs. If ``False`` error, warnings and infos will be
+        logged.
+    """
+    try:
+        log = logging.getLogger()
```

Comment:
I have to admit I don't know *everything* about the `logging` library. But, is this `log` the same as the other global ones?

My practical question is. If we want to log inside the functions in `utils.py` (or other modules), which `log` should we call?

Suggestion. We could pass the `log` object as an argument of the `setup_logging()`. Please correct me if I am not seeing the whole picture, but note the example:

```python
>>> import logging
>>> log = logging.getLogger(__name__)
>>> log2 = logging.getLogger()
>>> log is log2
False
>>> log3 = logging.getLogger(__name__)
>>> log is log3
True
>>> 
```

Reply:
I think the `setup_loger` is collecting every log message regardless how its called...?

Reply:
Yes. But my mind blows with the intrinsic intricacy of the `logging` library.
What about: 

```python
# we would send the log from the cli.py
def setup_logging(logobj, logfile=None, debug=False):
```
* then, remove line `41`
* and in line `64` change `logger` to `log`?

What is the difference between the `logger` object and the `log` ? I am not saying it is wrong, simply I can't see it. Is it all the same?

Very nice work :tada: :clap: